src/TaggedInput.jsx

```diff
@@ -215,18 +217,20 @@ var TaggedInput = React.createClass({
     switch (e.keyCode) {
       case KEY_CODES.BACKSPACE:
         if (!e.target.value || e.target.value.length < 0) {
-          if (p.onBeforeRemoveTag(s.tags.length - 1)) {
-            poppedValue = s.tags.pop();
+          if (!p.onBeforeRemoveTag(s.tags.length - 1)) {
```

Comment:
Even here, can you move it inside the if.
I am gonna do a little refactor to make the code a bit more readable and less verbose after this PR is in.